framework-core/src/main/java/org/daisy/pipeline/job/impl/StatusResultProvider.java

```diff
@@ -0,0 +1,33 @@
+package org.daisy.pipeline.job.impl;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.transform.Result;
+import javax.xml.transform.stream.StreamResult;
+
+import com.google.common.base.Supplier;
+import com.google.common.collect.Iterables;
+
+public final class StatusResultProvider implements Supplier<Result> {
+
+	public StatusResultProvider(String port) {
+		portName = port;
+	}
+
+	private final String portName;
+	private List<ByteArrayOutputStream> results = new ArrayList<>();
```

Comment:
minor comment: maybe `LinkedList` would be slightly more adapted here?

Reply:
This is indeed what DynamicResultProvider uses. You mean it's better because we don't need random access?

Reply:
> You mean it's better because we don't need random access?

Yes; we’re only adding to the list, or converting it as a whole. That said, the performance gain or any other impact should be very marginal, it’s just that LinkedList sounds like a more natural fit.